tests/integration/traces/test_traces_command.py

```diff
@@ -0,0 +1,223 @@
+import time
+from pathlib import Path
+from unittest import skipIf
+
+import boto3
+from tests.integration.deploy.deploy_integ_base import DeployIntegBase
+from tests.integration.package.package_integ_base import PackageIntegBase
+from tests.integration.traces.traces_integ_base import TracesIntegBase, RETRY_COUNT
+from tests.testing_utils import run_command, RUNNING_ON_CI, RUNNING_TEST_FOR_MASTER_ON_CI, RUN_BY_CANARY
+from datetime import datetime
+import logging
+from parameterized import parameterized
+
+LOG = logging.getLogger(__name__)
+
+SKIP_TRACES_TESTS = RUNNING_ON_CI and RUNNING_TEST_FOR_MASTER_ON_CI and not RUN_BY_CANARY
+
+
+@skipIf(SKIP_TRACES_TESTS, "Skip traces tests in CI/CD only")
+class TestTracesCommand(TracesIntegBase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.lambda_client = boto3.client("lambda")
+        cls.cfn_client = boto3.client("cloudformation")
+        cls.xray_client = boto3.client("xray")
+
+        cls.test_data_path = Path(__file__).resolve().parents[1].joinpath("testdata", "traces")
+
+        package = PackageIntegBase()
+        cls.stack_name = package._method_to_stack_name("test.integration.test_traces_command")
+
+        LOG.info("Deploying stack %s", cls.stack_name)
+        deploy = DeployIntegBase()
```

Comment:
Same goes here, we should not be instantiating test classes inside another one. If you need to use instance methods, please check https://github.com/aws/aws-sam-cli/pull/3607 or https://github.com/aws/aws-sam-cli/pull/3580

Reply:
Create a simply get deploy command list method in traces integration base class to avoid this issue.

Reply:
updated in the latest commit.